checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakTransfer.java

```diff
@@ -0,0 +1,139 @@
+public class ResourceLeakTransfer extends CalledMethodsTransfer {
+
+  /**
+   * Shadowed because we MUST dispatch to the RLC's version of getTypefactoryOfSubchecker to get the
+   * correct MCATF.
+   */
+  private final ResourceLeakAnnotatedTypeFactory rlTypeFactory;
+
+  /**
+   * Create a new RL transfer function.
+   *
+   * @param analysis the analysis. Note that the associated type factory must be a Resource Leak
+   *     type factory.
+   */
+  public ResourceLeakTransfer(final CFAnalysis analysis) {
+    super(analysis);
+    this.rlTypeFactory = (ResourceLeakAnnotatedTypeFactory) analysis.getTypeFactory();
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitTernaryExpression(
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitMethodInvocation(
+      final MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+
+    TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(node, input);
+
+    handleCreatesMustCallFor(node, result);
+    updateStoreWithTempVar(result, node);
+
+    Node receiver = node.getTarget().getReceiver();
+    // If there is a temporary variable for the receiver, update its type.
+    MustCallAnnotatedTypeFactory mcAtf =
+        rlTypeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+    Node accumulationTarget = mcAtf.getTempVar(receiver);
+
+    if (accumulationTarget != null) {
+      String methodName = node.getTarget().getMethod().getSimpleName().toString();
+      methodName = rlTypeFactory.adjustMethodNameUsingValueChecker(methodName, node.getTree());
+      accumulate(accumulationTarget, result, methodName);
+    }
+
+    return result;
+  }
+
+  /**
+   * Clears the called-methods store of all information about the target if an @CreatesMustCallFor
+   * method is invoked and the type factory can create obligations. Othewise, does nothing.
+   *
+   * @param n a method invocation
+   * @param result the transfer result whose stores should be cleared of information
+   */
+  private void handleCreatesMustCallFor(
+      MethodInvocationNode n, TransferResult<CFValue, CFStore> result) {
+    if (!rlTypeFactory.canCreateObligations()) {
+      return;
+    }
+
+    List<JavaExpression> targetExprs =
+        CreatesMustCallForElementSupplier.getCreatesMustCallForExpressions(
+            n, rlTypeFactory, rlTypeFactory);
+    for (JavaExpression targetExpr : targetExprs) {
+      AnnotationMirror defaultType = rlTypeFactory.top;
+      if (result.containsTwoStores()) {
+        CFStore thenStore = result.getThenStore();
+        thenStore.clearValue(targetExpr);
+        thenStore.insertValue(targetExpr, defaultType);
```

Comment:
Use `replaceValue` rather than `clearValue` then `insertValue`.  Then you won't need all the local variables, either.

Reply:
@kelloggm Ping.